drivers/i2c/Kconfig

```diff
@@ -90,19 +90,36 @@ config I2C_NRF5_GPIO_SCL_PIN
 	help
 	  The GPIO pin to use for SCL.
 
-config I2C_STM32LX
-	bool "STM32Lx MCU I2C Driver"
-	depends on SOC_FAMILY_STM32 && SOC_SERIES_STM32L4X
+config I2C_STM32
+	bool "STM32 MCU I2C Driver"
+	depends on SOC_FAMILY_STM32
+        depends on I2C_STM32_V1 || I2C_STM32_V2
+	default n
+	help
+	  Enable I2C support on the STM32 family of processors.
+
+config I2C_STM32_INTERRUPT
+	bool "STM32 MCU I2C Interrupt Support"
+	depends on I2C_STM32
 	default n
 	help
-	  Enable I2C support on the STM32Lxx family of processors.
+	  Enable Interrupt support for the I2C Driver
 
-config I2C_STM32LX_INTERRUPT
-	bool "STM32Lx MCU I2C Interrupt Support"
-	depends on I2C_STM32LX
+config I2C_STM32_V1
+	bool "STM32 V1 Driver"
```

Comment:
the symbol name is fine, in the text we can do"

bool "STM32 V1 Driver - Used on F1 & F4X SoCs"
...